marshal/claim.go

```diff
@@ -229,32 +239,18 @@ func (c *claim) GetCurrentLag() int64 {
 	return 0
 }
 
-// Release will invoke our release mechanism if and only if we are still claimed.
-func (c *claim) Release() bool {
-	if !atomic.CompareAndSwapInt32(c.claimed, 1, 0) {
+// Release will invoke commit offsets and may release the Kafka partition depending on whether
+// releasePartition was passed or not
+func (c *claim) Release(releasePartition bool) bool {
```

Comment:
I think the interface to Release should be that it releases the partition. I'm fine with having a Terminate method that can be used to release or not release, but I think a method called Release should always do what it says on the tin.